lib/vagrant/plugin/v2/command.rb

```diff
@@ -230,6 +230,10 @@ def with_target_vms(names=nil, options=nil)
           color_index = 0
 
           machines.each do |machine|
+            if machine.state && machine.state.id != :not_created && !@env.machine_index.include?(machine.id)
```

Comment:
The last part of this if-statement needs to be updated. I had a running machine that existed in the global machine index, but it kept being called to recover despite being in the json file already. The actual id `machine.id` doesn't match with how we store it in the global index file, so that check will always be true and we'll always recover:

```ruby
[2] pry(#<VagrantPlugins::CommandStatus::Command>)> @env.machine_index
=> #<Vagrant::MachineIndex:0x0000561c5563fb38
 @data_dir=#<Pathname:/home/brian/.vagrant.d/data/machine-index>,
 @index_file=#<Pathname:/home/brian/.vagrant.d/data/machine-index/index>,
 @lock=
  #<Monitor:0x0000561c5563ed50
   @mon_count=0,
   @mon_mutex=#<Thread::Mutex:0x0000561c5563eaf8>,
   @mon_mutex_owner_object_id=47339845842600,
   @mon_owner=nil>,
 @machine_locks={},
 @machines=
  {"a8756e374a7745c08aac72797a60522e"=>
    {"local_data_path"=>"/home/brian/code/vagrant-sandbox/vagrant-installers/.vagrant",
     "name"=>"ubuntu-14.04",
     "provider"=>"vmware_desktop",
     "state"=>"running",
     "vagrantfile_name"=>nil,
     "vagrantfile_path"=>"/home/brian/code/vagrant-sandbox/vagrant-installers",
     "updated_at"=>nil,
     "extra_data"=>
      {"box"=>
        {"name"=>"hashicorp-vagrant/ubuntu-14.04", "provider"=>"vmware_desktop", "version"=>"1.0.20180822143433"}}},
   "ae12c6cd2b7a4d0f8c3c758e2db80570"=>
    {"local_data_path"=>"/home/brian/code/vagrant-sandbox/.vagrant",
     "name"=>"bork",
     "provider"=>"virtualbox",
     "state"=>"running",
     "vagrantfile_name"=>nil,
     "vagrantfile_path"=>"/home/brian/code/vagrant-sandbox",
     "updated_at"=>nil,
     "extra_data"=>
      {"box"=>{"name"=>"elastic/ubuntu-16.04-x86_64", "provider"=>"virtualbox", "version"=>"20200607.0.0"}}}}>
[9] pry(#<VagrantPlugins::CommandStatus::Command>)> !@env.machine_index.include?(machine.id)
=> true
[10] pry(#<VagrantPlugins::CommandStatus::Command>)> machine.id
=> "f6ded812-5402-44ec-956e-8493109ed32e"
```

I'm not sure why there's a difference here :thinking:

Reply:
I think it's because `machine.id` != the machines uuid?  I think maybe we'd want `machine.index_uuid` here

Reply:
This looks right:

```ruby
[10] pry(#<VagrantPlugins::CommandStatus::Command>)> machine.id
=> "f6ded812-5402-44ec-956e-8493109ed32e"
[11] pry(#<VagrantPlugins::CommandStatus::Command>)> machine.index_uuid
=> "ae12c6cd2b7a4d0f8c3c758e2db80570"
```

Reply:
good catch, thanks!